fetch.bs

```diff
@@ -1037,6 +1027,138 @@ worked on in <a href=https://github.com/whatwg/fetch/issues/1156>issue #1156</a>
+  <var>body</var>'s <a for=body>stream</a>.
+
+  <p class="note no-backref">This operation cannot throw an exception.
+
+ <li><p>Perform the <a>incrementally-read loop</a> given <var>reader</var>,
+ <var>taskDestination</var>, <var>processBodyChunk</var>, <var>processError</var>, and
+ <var>processEndOfBody</var>.
+</ol>
+
+<p>To perform the <dfn>incrementally-read loop</dfn>, given a {{ReadableStreamDefaultReader}} object
+<var>reader</var>, <a for=/>parallel queue</a> or <a for=/>global object</a>
+<var>taskDestination</var>, algorithm <var>processBodyChunk</var>, algorithm
+<var>processError</var>, and null or algorithm <var>processEndOfBody</var>:
+
+<ol>
+ <li>
+  <p>Let <var>readRequest</var> be the following <a>read request</a>:
+
+  <dl>
+   <dt><a for="read request">chunk steps</a>, given <var>chunk</var>
+   <dd>
+    <ol>
+     <li><p>Let <var>continueAlgorithm</var> be null.
+
+     <li><p>If <var>chunk</var> is not a {{Uint8Array}} object, then set
+     <var>continueAlgorithm</var> to this step: run <var>processError</var> given a {{TypeError}}.
+
+     <li>
+      <p>Otherwise:
+
+      <ol>
+       <li><p>Let <var>bytes</var> be a <a lt="get a copy of the buffer source">copy of</a>
+       <var>chunk</var>.
+
+       <li>
+        <p>Set <var>continueAlgorithm</var> to these steps:
+
+        <ol>
+         <li><p>Run <var>processBodyChunk</var> given <var>bytes</var>.
+
+         <li><p>Perform the <a>incrementally-read loop</a> given <var>reader</var>,
+         <var>taskDestination</var>, <var>processBodyChunk</var>, <var>processError</var>, and
+         <var>processEndOfBody</var>.
+        </ol>
+      </ol>
+
+     <li><p><a>Queue a fetch task</a> given <var>continueAlgorithm</var> and
+     <var>taskDestination</var>.
+    </ol>
+
+   <dt><a for="read request">close steps</a>
+   <dd><ol><li><p>If <var>processEndOfBody</var> is non-null, then <a>queue a fetch task</a> given
+   <var>processEndOfBody</var> and <var>taskDestination</var>.</ol>
+
+   <dt><a for="read request">error steps</a>, given <var>e</var>
+   <dd>
+    <ol>
+     <li><p>Let <var>errorAlgorithm</var> be this step: run <var>processError</var> given
+     <var>e</var>.
+
+     <li><p><a>Queue a fetch task</a> given <var>errorAlgorithm</var> and
+     <var>taskDestination</var>.
+    </ol>
+  </dl>
+
+ <li><p><a for=ReadableStreamDefaultReader>Read a chunk</a> from <var>reader</var> given
+ <var>readRequest</var>.
+</ol>
+
+<p>To <dfn export for=body>fully read</dfn> a <a for=/>body</a> <var>body</var>, given an algorithm
+<var>processBody</var>, an optional null or algorithm
+<dfn export for="body/fully read"><var>processError</var></dfn> (default null), and an optional
+null, <a for=/>parallel queue</a>, or <a for=/>global object</a>
+<dfn export for="body/fully read">taskDestination</dfn> (default null), run these steps.
+<var>processBody</var> must be an algorithm accepting failure or a <a for=/>byte sequence</a>. If
```

Comment:
I think it doesn't need to handle the failure case; processError deals with that?